### PR TITLE
Address issues identified in UX pass

### PR DESCRIPTION
**Pull request by jrbotros**

### CODE REVIEW

File: orchestra/static/orchestra/timing/timecard/controller.js [hunk 1]
```diff
@@ -74,7 +75,16 @@
...
         }
         entry.description = entry.editData.description;
-        entry.time_worked = moment.duration(entry.editData.timeWorked);
+
+        // Only truncate seconds worked if the user has changed things
+        if (!angular.equals(entry.editData.timeWorked, entry.time_worked.componentize())) {
+          entry.time_worked = moment.duration(entry.editData.timeWorked);
+        }
+
+        if (!entry.editData.date.isSame(entry.date)) {
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timecard/controller.js:85]: When I create a new entry, it doesn't auto-fill the date, so isSame throws a null exception

File: orchestra/static/orchestra/timing/timecard/partials/timecard.html [hunk 2]
```diff
@@ -2,107 +2,120 @@
...
+                  </div>
                 </div>
               </div>
-            </div>
+              <div class="entry-row" ng-class="{'gray-stripe': $index % 2}" ng-repeat="entry in dateEntries | orderBy:'-id'">
+                <form class="container-fluid" novalidate name="entryForm" class="form-inline">
+                  <div class="row">
+                    <div class="col-xs-2 edit-tools disable-select" ng-if="!vm.expiredEntry(entry)">
+                      <i ng-click="vm.timeEntries.deleteEntry(entry)" class="fa fa-times-circle"></i>
+                      <i class="fa fa-pencil-square-o" ng-class="{active: entry.editing}" ng-click="vm.editEntry(entry)"></i>
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timecard/partials/timecard.html:37]: hide button vs. deactivate?

File: orchestra/static/orchestra/timing/timecard/controller.js [hunk 3]
```diff
@@ -63,9 +56,17 @@
         }
       };
 
+      vm.addEntry = function(date) {
+        vm.timeEntries.createEntry(date).then(function(entry) {
+          entry.editing = true;
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timecard/controller.js]: set entry.editData.date to date? i think that would address the isSame null exception above

File: orchestra/static/orchestra/dashboard/js/controllers.js [hunk 4]
```diff
@@ -20,6 +20,13 @@
       vm.waiting = false;
     });
 
+    vm.newTask = function(taskType) {
```
Comment (lydiagu) [inline orchestra/static/orchestra/dashboard/js/controllers.js]: why is newTask defined again? Also defined on line 32

File: orchestra/static/orchestra/timing/service.js [hunk 5]
```diff
@@ -46,6 +46,27 @@
...
         },
 
+
+        /**
+         * Creates a new TimeEntry server-side and initializes with returned data.
+         */
+        createEntry: function(date) {
+          var service = this;
+
+          var createUrl = '/orchestra/api/interface/time_entries/';
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/service.js]: you didn't have a createEntry function before?

Comment (lydiagu) [inline orchestra/static/orchestra/timing/service.js]: oh i see, it's moved from the other service

**Comment by lydiagu** [comment]
Some issue with width of timer icon
image

---

@jrbotros ✅

File: orchestra/static/orchestra/timing/timecard/controller.js [hunk 6]
```diff
@@ -74,7 +75,16 @@
           return;
         }
         entry.description = entry.editData.description;
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/timecard/controller.js:78]: i get a TypeError: Cannot read property 'description' of undefined at this line when i try to save an empty entry

**Comment by lydiagu** [comment]
missing merge migration

done reviewing! yay almost done!

File: orchestra/static/orchestra/timing/timecard/controller.js (hunk 1 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js:85]: Oops, now it's set to auto-fill the date.

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js:85]: ✅

File: orchestra/static/orchestra/timing/timecard/partials/timecard.html (hunk 2 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/partials/timecard.html:37]: I don't think the button should mysteriously disappear, I think that would look weird. Going to leave as is.

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/partials/timecard.html:37]: ✅

File: orchestra/static/orchestra/timing/timecard/controller.js (hunk 3 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js]: This is fixed along with your first comment.

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js]: ✅

File: orchestra/static/orchestra/dashboard/js/controllers.js (hunk 4 above)
Comment (jrbotros) [inline orchestra/static/orchestra/dashboard/js/controllers.js]: Oops, fixed ✅

File: orchestra/static/orchestra/timing/timecard/controller.js (hunk 6 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js:78]: This should be good, now!

Comment (jrbotros) [inline orchestra/static/orchestra/timing/timecard/controller.js:78]: ✅

**Pull request merged**